petastorm/unischema.py

```diff
@@ -234,9 +234,10 @@ def from_arrow_schema(cls, parquet_dataset):
         for column_name in arrow_schema.names:
             arrow_field = arrow_schema.field_by_name(column_name)
             field_type = arrow_field.type
-            codec, np_type = _numpy_and_codec_from_arrow_type(field_type)
+            if not isinstance(field_type, ListType):
```

Comment:
I think this discards valid cases that are supported. That's the reason for failure of the tests. We should be more precise here. Would also consider to issue a `warning.warn` for the user not to be surprised. Would need to cover this case with a unit test.

Please let us know if you plan to continue working on it (would be more than happy to get this as a contribution). Alternatively, we can help resolve this case.

Thanks!

Reply:
Good idea, I'll add this warning.
What do you mean by more precise?

Thanks for the proposal, I still can allocate time to work on it so I will continue for now. I'll let you know if needed.